libraries/azure_management_group.rb

```diff
@@ -0,0 +1,130 @@
+require 'azure_generic_resource'
+
+class AzureManagementGroup < AzureGenericResource
+  name 'azure_management_group'
+  desc 'Verifies settings for an Azure Management Group'
+  example <<-EXAMPLE
+    describe azure_management_group(group_id: 'example-group') do
+      it { should exist }
+    end
+  EXAMPLE
+
+  def initialize(opts = {})
+    # Options should be Hash type. Otherwise Ruby will raise an error when we try to access the keys.
+    raise ArgumentError, 'Parameters must be provided in an Hash object.' unless opts.is_a?(Hash)
+
+    opts[:resource_provider] = specific_resource_constraint('Microsoft.Management/managementGroups', opts)
+    opts[:resource_uri] = "providers/#{opts[:resource_provider]}"
+    opts[:add_subscription_id] = false
+    opts[:resource_identifiers] = %i(group_id)
+    opts[:allowed_parameters] = %i(expand recurse filter)
+    # For backward compatibility.
+    opts[:query_parameters] = {
+      '$recurse' => opts[:recurse] || false,
+    }
+    opts[:query_parameters].merge!('$expand' => opts[:expand]) unless opts[:expand].nil?
+    # Note that $expand=children must be passed up if $recurse is set to true.
```

Comment:
This code answers my question... I think the wording should be changed slightly. `$expand=children must be set if $recurse is set to true.`
OR
`$expand=children will be set if $recurse is set to true.`

Reply:
could this comment also be updated for consistency?